src/Components/TodoApp.js

```diff
@@ -0,0 +1,113 @@
+
+import TodoItem from './TodoItem';
+import Input from "./Input";
+import Footer from "./Footer";
+import Button from "./Button";
+import { useCallback, useMemo } from "react";
+import Box from "@mui/material/Box";
+import { useSelector, useDispatch } from "react-redux";
+import {
+  addTodo,
+  checkAll,
+  toggleDone,
+  edit,
+  replace,
+  deleteTodo,
+  clear,
+  newTodo,
+  editTodo,
+  clickFilter,
+} from "./actions/index";
+
+function TodoApp(){
+
+   
+    const initial = useSelector((state) => state.todoReducers);
+    const { all, mode, newToDo, editToDo } = initial;
+  
+    //to dispatch action
+    const dispatch = useDispatch();
+  
+    //giving filtered array
+    const { listToMap, list, completed, active } = useMemo(() => {
+      const activeArray = [...all].filter((item) => item.done === false);
+      const completeArray = [...all].filter((item) => item.done === true);
+  
+      return {
+        listToMap:
+          mode === "active"
+            ? activeArray
+            : mode === "completed"
+            ? completeArray
+            : all,
+        list: all.length,
+        completed: completeArray.length,
```

Comment:
no need to keep separate counts for all, just count the number of todos in the listToMap array directly in the UI